File: 120-triangle/triangle.c
int minimum(int arr[], int size) { 
    int min = arr[0]; 
    for (int i = 1; i < size; i++) { 
        if (arr[i] < min) { 
            min = arr[i]; 
        } 
    } 
    return min; 
} 
int minimumTotal(int** triangle, int triangleSize, int* triangleColSize) {
    for (int i = triangleSize - 2; i >= 0; i--) {
        for (int j = 0; j <= i; j++) {
            if (triangle[i + 1][j] < triangle[i + 1][j + 1]) {
                triangle[i][j] += triangle[i + 1][j];
            } else {
                triangle[i][j] += triangle[i + 1][j + 1];
            }
        }
    }
    return triangle[0][0];
}

File: 120-triangle/triangle.py
class Solution(object):
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        def minimum(arr, size):
            min_value = arr[0]
            for i in range(1, size):
                if arr[i] < min_value:
                    min_value = arr[i]
            return min_value
        for i in range(len(triangle) - 2, -1, -1):
            for j in range(i + 1):
                triangle[i][j] += min(triangle[i + 1][j], triangle[i + 1][j + 1])
        return triangle[0][0]